hive/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -155,14 +164,146 @@ public void renameTable(TableIdentifier from, TableIdentifier to) {
       throw new org.apache.iceberg.exceptions.AlreadyExistsException("Table already exists: %s", to);
 
     } catch (TException e) {
-      throw new RuntimeException("Failed to rename " + from.toString() + " to " + to.toString(), e);
+      throw new RuntimeException("Failed to rename " + from + " to " + to, e);
 
     } catch (InterruptedException e) {
       Thread.currentThread().interrupt();
       throw new RuntimeException("Interrupted in call to rename", e);
     }
   }
 
+  @Override
+  public void createNamespace(Namespace namespace, ImmutableMap<String, String> meta) {
+    Preconditions.checkArgument(
+        !namespace.isEmpty(),
+        "Cannot create namespace with invalid name: %s", namespace);
+    Preconditions.checkArgument(
+        namespace.levels().length == 1,
+        "Cannot support multi part namespace in Hive MetaStore: %s", namespace);
+
+    try {
+      clients.run(client -> {
+        client.createDatabase(convertToDatabase(namespace, meta));
+        return null;
+      });
+
+    } catch (AlreadyExistsException e) {
+      throw new org.apache.iceberg.exceptions.AlreadyExistsException(e, "Namespace '%s' already exists!",
+            namespace);
+
+    } catch (TException e) {
+      throw new RuntimeException("Failed to create namespace " + namespace + " in Hive MataStore", e);
+
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw new RuntimeException(
+          "Interrupted in call to createDatabase(name) " + namespace + " in Hive MataStore", e);
+    }
+  }
+
+  @Override
+  public List<Namespace> listNamespaces(Namespace namespace) {
+    if (namespace.levels().length > 1) {
+      throw new NoSuchNamespaceException("Namespace does not exist: %s", namespace);
+    }
+    if (!namespace.isEmpty()) {
+      return ImmutableList.of();
+    }
+    try {
+      return clients.run(
+          HiveMetaStoreClient::getAllDatabases)
+          .stream()
+          .map(Namespace::of)
+          .collect(Collectors.toList());
+
+    } catch (TException e) {
+      throw new RuntimeException("Failed to list all namespace: " + namespace + " in Hive MataStore",  e);
+
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw new RuntimeException(
+          "Interrupted in call to getAllDatabases() " + namespace + " in Hive MataStore", e);
+    }
+  }
+
+  @Override
+  public boolean dropNamespace(Namespace namespace) {
+    if (namespace.levels().length != 1) {
+      throw new NoSuchNamespaceException("Namespace does not exist: %s", namespace);
+    }
+
+    try {
+      clients.run(client -> {
+        client.dropDatabase(namespace.level(0),
+            false /* deleteData */,
+            false /* ignoreUnknownDb */,
+            true /* cascade */);
```

Comment:
Why does this cascade, but not delete data? Can you explain these options?

Reply:
OK I fixed it delete data together at here, I was think delete data is a danger operation.

Reply:
Can you explain what all 3 options do? I'm not sure that this should cascade using Hive. If we are going to delete all tables, then this should probably be implemented in Iceberg so that we remove files that aren't in the table root.

Reply:
`deleteData `delete the underlying data or just delete the table in metadata store, we only delete the metadata from meta store, so `deleteData` is false.
 `cascade` is true,  while drop the database, this will drop the table of the database one by one, before drop the database.

Reply:
I think we need to decide what the semantics are for `dropNamespace`. There are a few options:

1. `dropNamespace` could check that the namespace is empty and throw a `NamespaceNotEmptyException` if it is not.
2. `dropNamespace` could recursively delete, which I think is what `cascade` would do
3. `dropNamespace` could have a `cascade` flag to recursively delete

I think the easiest option to get this PR in is to go with Option 1. We can always add a `cascade` flag later to go switch to Option 3. And that way we don't have to worry about Hive behaviors and how to correctly clean up tables right now.

Reply:
OK thanks @rdblue I have implement the `dropNamespace` method like Option 1.

Reply:
Thank you! It looks much better. Just a few more comments.